integrations/oci/metrics/metrics/src/main/java/io/helidon/integrations/oci/metrics/OciMetricsSupport.java

```diff
@@ -0,0 +1,424 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.helidon.integrations.oci.metrics;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+import io.helidon.config.Config;
+import io.helidon.config.metadata.ConfiguredOption;
+import io.helidon.metrics.api.RegistryFactory;
+import io.helidon.webserver.Routing;
+import io.helidon.webserver.Service;
+import io.helidon.webserver.WebServer;
+
+import com.oracle.bmc.monitoring.Monitoring;
+import com.oracle.bmc.monitoring.model.MetricDataDetails;
+import com.oracle.bmc.monitoring.model.PostMetricDataDetails;
+import com.oracle.bmc.monitoring.requests.PostMetricDataRequest;
+import org.eclipse.microprofile.metrics.Metadata;
+import org.eclipse.microprofile.metrics.MetricID;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics.MetricRegistry.Type;
+import org.eclipse.microprofile.metrics.MetricType;
+import org.eclipse.microprofile.metrics.MetricUnits;
+
+/**
+ * OCI Metrics Support
+ * <p>
+ * Because this service does not create an endpoint, a calling SE app does not need to add this service to its routing rules.
+ * But the caller does need to invoke {@link #update(Routing.Rules)} and pass the routing rules so, when the
+ * web server shuts down, this service can shut down in an orderly way.
```

Comment:
I supplied this comment myself, and I should have revised it earlier. The comment should say something like this:
```
 * <p>
 * Even though this service does not create an endpoint, a calling SE app should still register it by invoking 
 * {@link Routing.Rules#register(io.helidon.webserver.Service...)}, as it should with any service it uses. That allows this service
 * to detect when the webserver shuts down so it can shut down as well.
 * </p>
```

Reply:
Updated in the most recent commit.